CMakeLists.txt

```diff
@@ -199,6 +199,7 @@ target_link_libraries(xtensor INTERFACE xtl)
 
 OPTION(XTENSOR_ENABLE_ASSERT "xtensor bound check" OFF)
 OPTION(XTENSOR_CHECK_DIMENSION "xtensor dimension check" OFF)
+OPTION(XTENSOR_FORCE_TEMPORARY_MEMORY_IN_ASSIGNMENTS "xtensor force the use of temporary memory when assigning instead of an automatic overlap check" ON)
```

Comment:
I'm not a fan of having different assign behaviors depending on an option, when it's not for managing dependency or debugging. We should probably have some pattern to dynamically decide if we should check for memory overlap or not. This can be done in a dedicated PR, as it may not be obvious to get it done correctly.

Reply:
So should we keep the option for now?